src/utils.c

```diff
@@ -1,11 +1,16 @@
 #include "data.table.h"
 
+bool inside_int32_range(double x) {
+  // N.B. if x = 2147483647.99 then (int)2147483647.99 is not undefined behaviour
+  return x < 2147483648 && x > -2147483648;
```

Comment:
isn't there a macro INT_MAX (and maybe INT_MIN) that would be preferable here for robustness?

Reply:
It would be `x < ((double)INT_MAX + 1.0)` etc which seemed a bit messy, plus I think `int` is guaranteed to be 32-bits in R, even if it could technically be embedded.

Reply:
It feels that 2147483647.01 is already out of range. Formula you used above in gh comment could go into code comment as well.

Reply:
jangorecki, is your objection that the logic is incorrect or that the function is poorly named? I'm happy to amend either, though as the comment notes 2147483647.01 is safe to coerce to `int` so that is the basis for the logic. Once I've clarified that I'm also happy to push through a more thorough comment.

Reply:
I have renamed the function to better reflect it is a test for whether the double can be represented as an int, not necessarily within its range.